pandas/tests/series/test_constructors.py

```diff
@@ -745,6 +745,25 @@ def test_constructor_signed_int_overflow_deprecation(self):
         expected = Series([1, 200, 50], dtype="uint8")
         tm.assert_series_equal(ser, expected)
 
+    @pytest.mark.parametrize(
+        "values",
+        [
+            np.array([1], dtype=np.uint16),
+            np.array([1], dtype=np.uint32),
+            np.array([1], dtype=np.uint64),
+            [np.uint16(1)],
+            [np.uint32(1)],
+            [np.uint64(1)],
+        ],
+    )
+    def test_constructor_numpy_uints(self, values):
```

Comment:
are pd.Index or pd.array affected?  pd.NumericIndex?

Reply:
pd.array and pd.NumericIndex are not impacted; I've added tests for Index.

```
arr = pd.array([np.uint16(1)])
print(arr)

# <IntegerArray>
# [1]
# Length: 1, dtype: Int64

index = NumericIndex([np.uint16(1)])
print(index)

# NumericIndex([1], dtype='uint16')
```

Both of these are the same as 1.4.3